### ISSUE TITLE
🚀 Feature: Bring in a no-redundant-publishConfig rule

### ISSUE THREAD

**Issue by azat-io** [JoshuaKGoldberg/eslint-plugin-package-json#1365]
### Feature Request Checklist

- [x] I have [searched for related issues](https://github.com/JoshuaKGoldberg/eslint-plugin-package-json/issues?q=is%3Aissue) and found none that matched my issue.

### Overview

Following #814: we need a `no-redundant-publishConfig` rule that warns when `publishConfig` has no effect.

Per [npm docs](https://docs.npmjs.com/cli/v8/commands/npm-publish), unscoped packages (without `@`) are always public. Using `publishConfig.access` in these packages is redundant since they cannot be made private.

The rule should:

- Warn when unscoped packages have `publishConfig.access` defined
- Remain silent for scoped packages (`@org/name`) where the field is meaningful

❌ Should warn:

```json
{
  "name": "my-package",
  "publishConfig": { "access": "public" }
}
```

✅ Should pass:

```json
{
  "name": "@myorg/my-package",
  "publishConfig": { "access": "public" }
}
```
